cwms-data-api/src/main/java/cwms/cda/data/dao/UnitsDao.java

```diff
@@ -12,4 +22,31 @@ public UnitsDao(DSLContext dsl) {
     public String getUnits(String format) {
         return CWMS_CAT_PACKAGE.call_RETRIEVE_UNITS_F(dsl.configuration(), format);
     }
+
+    public List<Unit> getUnits()
+    {
+        Map<Long, List<String>> unitIdToAliasMap = dsl.select()
```

Comment:
select cu.unit_id, cu.long_name, cu.unit_system, cu.description, cap.abstract_param_id, atu.alias_id
from cwms_20.cwms_unit cu
inner join cwms_20.at_unit_alias atu on (cu.unit_code = atu.unit_code and (db_office_code = 53 /*or db_office_code = SWT*/))
inner join cwms_20.cwms_abstract_parameter cap on (cu.abstract_param_code = cap.abstract_param_code)

Reply:
In my opinion, give AT_UNIT_ALIAS supports office specific aliases, that this query and unit object should include an office_id field. Correspondingly, you could limit the query to only include data from CWMS (53), or CWMS and a optionally parameterized office_id

Reply:
@RyanM-RMA